### PR TITLE
Use the room heroes when computing a DM avatar.

### LINKED ISSUE #1659: Tinted avatar can be different in the room setting screen
I saw the problem in a single DM. It may be a race condition: killing the app does not fix the problem but clearing the cache, yes.

In the room list and in the timeline:
IMG_8072

In the DM room:
IMG_8073

### PR DESCRIPTION
**Pull request by pixlwave**
Requires https://github.com/matrix-org/matrix-rust-sdk/pull/3503

This PR makes the following changes:
- Add heroes from the SDK to both `RoomSummaryDetails` and `RoomProxy`.
- Add a new `RoomAvatar` enum that contains either
    - The room's ID, name and avatar URL.
    - The room's heroes for a 1:1 DM where the avatar is nil.
- Add a new `RoomAvatarImage` and use this everywhere instead of `LoadableAvatarImage` for rooms.
- Move the stacked avatar layout from `AvatarHeaderView` into the `RoomAvatarImage` in preparation for clustering up to 4 hero avatars.

Fixes #1659

### CODE REVIEW

File: ElementX/Sources/Mocks/RoomProxyMock.swift [hunk 1]
```diff
@@ -54,6 +54,7 @@ extension RoomProxyMock {
         id = configuration.id
         name = configuration.name
         topic = configuration.topic
+        avatar = .room(id: configuration.id, name: configuration.name, avatarURL: configuration.avatarURL) // TODO: What happens for the logic in here?
```
Comment (stefanceriu) [inline ElementX/Sources/Mocks/RoomProxyMock.swift:57]: What's this about? 🤔

File: ElementX/Sources/Other/SwiftUI/Views/AvatarHeaderView.swift [hunk 2]
```diff
@@ -17,37 +17,18 @@
...
-        }
-        
-        init(from user: UserProfileProxy) {
-            id = user.userID
-            name = user.displayName
-            avatarURL = user.avatarURL
-        }
+    private enum AvatarInfo {
+        case room(RoomAvatar)
+        case user(UserProfileProxy)
```
Comment (stefanceriu) [inline ElementX/Sources/Other/SwiftUI/Views/AvatarHeaderView.swift:22]: 👌

**Review by stefanceriu — APPROVED** [review]
Very nice and great cleanup! 👏

File: ElementX/Sources/Mocks/RoomProxyMock.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Mocks/RoomProxyMock.swift]: Ah, I was unsure if it was worth copying the `RoomProxy` logic into the mock. I don't think it is, it won't prove anything on tests. I've replace it with a comment. Maybe one day we'll remove RoomProxyMock all together an use `RoomProxy` with an SDK `RoomMock`.

Comment (stefanceriu) [reply ElementX/Sources/Mocks/RoomProxyMock.swift]: Sounds like a plan 👌

**Pull request merged**